roles/resource_module/templates/module_utils/network_os/facts/base.py.j2

```diff
@@ -47,10 +46,11 @@ class FactsBase(object): #pylint: disable=R0205
         for key, val in iteritems(spec):
             if 'default' in val:
                 dct = {key: val['default']}
+            elif 'type' in val and val['type'] == 'dict':
+                dct = {key: self.generate_dict(val['options'])}
             else:
                 dct = {key: None}
             obj.update(dct)
-
         return obj
 
     @staticmethod
```

Comment:
Can you we not make the rest of the methods static here? Anyway `generate_final_config` is going to called with `self` in `resources/resources.py`.

Reply:
generate_dict and generate_final_config were changed because they call self, the others can stay static until they need self, if ever